src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  const dateParts = date.split(fromSeparator);
+
+  let year = '';
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dateParts[i];
+    } else if (fromFormat[i] === 'MM') {
+      month = dateParts[i];
+    } else if (fromFormat[i] === 'DD') {
+      day = dateParts[i];
+    }
+  }
+
+  if (toFormat.includes('YYYY') && fromFormat.includes('YY')) {
+    year = year < 30 ? `20${year}` : `19${year}`;
```

Comment:
The logic for converting 'YY' to 'YYYY' assumes that years less than 30 belong to the 2000s and others to the 1900s. This might not be accurate for all use cases. Consider making this logic configurable or providing a more robust solution.